### PR TITLE
22w03a stuff

### PR DESCRIPTION
**Pull request by apple502j**
Excluding: most rendering changes, commands, worldgen.

I know I screwed up 6856, give me a sec

### CODE REVIEW

**Comment by apple502j** [comment]
todo: Beehive generation field rename, FallingBlockEntity rename

File: mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping [hunk 1]
```diff
@@ -33,8 +33,11 @@ CLASS net/minecraft/class_4180 net/minecraft/world/storage/SerializingRegionBase
 		ARG 1 pos
 	METHOD method_19295 getOrCreate (J)Ljava/lang/Object;
 		ARG 1 pos
+	METHOD method_20366 (JZLjava/lang/Object;)V
+		ARG 4 sections
 	METHOD method_20368 update (Lnet/minecraft/class_1923;Lcom/mojang/serialization/DynamicOps;Ljava/lang/Object;)V
 		ARG 1 pos
+		ARG 2 dynamicOps
```
Comment (haykam821) [inline mappings/net/minecraft/world/storage/SerializingRegionBasedStorage.mapping]:
```suggestion
		ARG 2 ops
```

File: mappings/net/minecraft/world/level/storage/LevelStorage.mapping [hunk 2]
```diff
@@ -6,11 +6,13 @@ CLASS net/minecraft/class_32 net/minecraft/world/level/storage/LevelStorage
 	FIELD field_200 TIME_FORMATTER Ljava/time/format/DateTimeFormatter;
 	FIELD field_25020 GENERATOR_OPTION_KEYS Lcom/google/common/collect/ImmutableList;
 	FIELD field_31836 DEFAULT_ICON Ljava/lang/String;
+	FIELD field_36348 DATA_KEY Ljava/lang/String;
 	METHOD <init> (Ljava/nio/file/Path;Ljava/nio/file/Path;Lcom/mojang/datafixers/DataFixer;)V
 		ARG 1 savesDirectory
 		ARG 2 backupsDirectory
 		ARG 3 dataFixer
 	METHOD method_17926 createLevelDataParser (Lcom/mojang/serialization/DynamicOps;Lnet/minecraft/class_5359;)Ljava/util/function/BiFunction;
+		ARG 0 nbtOps
```
Comment (haykam821) [inline mappings/net/minecraft/world/level/storage/LevelStorage.mapping]:
```suggestion
		ARG 0 ops
```

File: mappings/net/minecraft/world/level/storage/LevelStorage.mapping [hunk 3]
```diff
@@ -58,6 +63,7 @@ CLASS net/minecraft/class_32 net/minecraft/world/level/storage/LevelStorage
 		METHOD method_27010 getDirectory (Lnet/minecraft/class_5218;)Ljava/nio/file/Path;
 			ARG 1 savePath
 		METHOD method_27013 readLevelProperties (Lcom/mojang/serialization/DynamicOps;Lnet/minecraft/class_5359;)Lnet/minecraft/class_5219;
+			ARG 1 nbtOps
```
Comment (haykam821) [inline mappings/net/minecraft/world/level/storage/LevelStorage.mapping]:
```suggestion
			ARG 1 ops
```

**Comment by apple502j** [comment]
Updates:

- Some classes related to NBT scanner are renamed, since generic words like "Query" and "Tree" will no longer be suitable after it's moved outside the inner class.
- "Tree" is renamed to "NbtTreeNode", since the instance itself is just a node.
- These two classes have Javadoc. Javadoc for record field bugged Enigma.
- `FallingBlockEntity#fromBlock` is renamed to `spawnFromBlock` since it spawns the entity as well.
- `NbtIo#read(DataInput, NbtScanner)` is now `NbtIo.scan` to follow the conventions.
- `SelectiveNbtCollector` javadoc now links to their sibling with completely opposite characters.
- Added Javadoc to `PacketListener#shouldCrashOnException`. This was worded to clear any confusion on when it will be used. If needed, I can also add that errors in Fabric Networking API will not cause crashes, but the paragraph on `client.execute` is enough IMO.
- `ServerPacketListener` uses the word "serverbound packet listeners". I am not sure if "C2S packet listener" or "serverside packet listener" is better.
- Misc javadocs. I tried to javadoc any method which could be useful for modders. I did not document most of TACS stuff, but maybe we can do later. "Persistent" tickets there mean tickets that do not expire, like player/dragon/forceload/portal, so that method keeps any tickets that are necessary for some processing while removing other tickets that can linger forever.

File: mappings/net/minecraft/entity/FallingBlockEntity.mapping [hunk 4]
```diff
@@ -20,6 +20,11 @@ CLASS net/minecraft/class_1540 net/minecraft/entity/FallingBlockEntity
 		ARG 2 pos
 	METHOD method_32879 (Lnet/minecraft/class_1282;FLnet/minecraft/class_1297;)V
 		ARG 2 entity
+	METHOD method_40005 spawnFromBlock (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;Lnet/minecraft/class_2680;)Lnet/minecraft/class_1540;
+		COMMENT Spawns a falling block entity at {@code pos} from the block {@code state}.
```
Comment (haykam821) [inline mappings/net/minecraft/entity/FallingBlockEntity.mapping:24]: This should document that the spawned entity is returned.

Comment (apple502j) [reply mappings/net/minecraft/entity/FallingBlockEntity.mapping:24]: @haykam821 Can I just write "and returns..."? Do I use `@returns`?

Comment (Juuxel) [reply mappings/net/minecraft/entity/FallingBlockEntity.mapping:24]: `@return the spawned entity` seems fine to me

Comment (apple502j) [inline mappings/net/minecraft/entity/FallingBlockEntity.mapping:24]:
```suggestion
		COMMENT Spawns a falling block entity at {@code pos} from the block {@code state}.
		COMMENT @return the spawned entity
```

**Review by Shnupbups — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**